src/vis/vis.js

```diff
@@ -338,45 +331,28 @@ var Vis = View.extend({
       this.mapView.bind('newLayerView', this.addTooltip, this);
     }
 
-    var cartoDBLayers;
-    var cartoDBLayerGroup;
-    layers = [];
+    var layers = [];
 
-    // This attribute is public (used by deep-insights)
-    this.interactiveLayers = new Backbone.Collection();
     _.each(data.layers, function (layerData) {
       if (layerData.type === 'layergroup' || layerData.type === 'namedmap') {
         var layersData;
-        var layerGroupClass;
         if (layerData.type === 'layergroup') {
           layersData = layerData.options.layer_definition.layers;
-          layerGroupClass = CartoDBLayerGroupAnonymous;
         } else {
           layersData = layerData.options.named_map.layers;
-          layerGroupClass = CartoDBLayerGroupNamed;
         }
-        cartoDBLayers = _.map(layersData, function (layerData) {
-          var cartoDBLayer = Layers.create('cartodb', self, layerData);
-          self.interactiveLayers.add(cartoDBLayer);
-          return cartoDBLayer;
-        });
-
-        cartoDBLayerGroup = new layerGroupClass(null, {
-          layers: cartoDBLayers
+        _.each(layersData, function (layerData) {
+          layers.push(Layers.create('CartoDB', self, layerData));
         });
-        layers.push(cartoDBLayerGroup);
       } else {
-        // Treat differently since this kind of layer is rendered client-side (and not through the tiler)
-        var layer = Layers.create(layerData.type, self, layerData);
-        layers.push(layer);
-        if (layerData.type === 'torque') {
-          self.interactiveLayers.add(layer);
-        }
+        layers.push(Layers.create(layerData.type, self, layerData));
       }
     });
 
-    // Map layers are resetted and the mapView adds the layers to the map
-    this.map.layers.reset(layers);
+    // TODO: This is PUBLIC. Remove dependency on this attribute from deep-insights.js
+    this.interactiveLayers = new Backbone.Collection(_.select(layers, function (layer) {
+      return layer.get('type') === 'CartoDB' || layer.get('type') === 'torque';
+    }));
```

Comment:
I think we need to rethink this `interactiveLayers` object, it will be invalid as soon as a user add/change/remove a layer, but probably better to tackle that in a separate issue/PR.